webapp/channels/src/components/new_search/select_team.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentTeamId, getMyTeams} from 'mattermost-redux/selectors/entities/teams';
+
+import {getCurrentLocale} from 'selectors/i18n';
+
+import * as Menu from 'components/menu';
+
+import './select_team.scss';
+import {Preferences} from 'utils/constants';
+import {filterAndSortTeamsByDisplayName} from 'utils/team_utils';
+
+import type {GlobalState} from 'types/store';
+
+interface Props {
+    value: string;
+    onChange: (team: string) => void;
+}
+
+const SelectTeam = (props: Props) => {
+    const intl = useIntl();
+    const myTeams = useSelector(getMyTeams);
+    const locale = useSelector(getCurrentLocale);
+    const userTeamsOrderPreference = useSelector((state: GlobalState) => get(state, Preferences.TEAMS_ORDER, '', ''));
+    const currentTeamId = useSelector(getCurrentTeamId);
+    const teams = filterAndSortTeamsByDisplayName(myTeams, locale, userTeamsOrderPreference);
+
+    const isDifferentTeamSelected = props.value !== currentTeamId;
+
+    const allTeamsText = intl.formatMessage({
+        id: 'search_teams_selector.all_teams',
+        defaultMessage: 'All Teams',
+    });
+
+    let menuButtonText = allTeamsText;
+    if (props.value) {
+        const team = teams.find((t) => t.id === props.value);
+        if (team) {
+            menuButtonText = team.display_name;
+        }
+    }
+
+    const button = (
+        <span>{menuButtonText}  <i className='icon icon-chevron-down'/></span>
+    );
+
+    return (
+        <Menu.Container
+            menuButton={{
+                id: 'searchTeamsSelectorMenuButton',
+                class: classNames('search-teams-selector-menu-button', {'search-teams-selector-menu-button__different-team': isDifferentTeamSelected}),
+                children: button,
+                as: 'span',
+            }}
+            menu={{
+                id: 'searchTeamSelectorMenu',
+            }}
```

Comment:
I think right-anchored would be more appropriate here

```suggestion
            }}
            anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'right',
            }}
            transformOrigin={{
                vertical: 'top',
                horizontal: 'right',
            }}
```
![CleanShot 2024-12-04 at 16 03 04](https://github.com/user-attachments/assets/6fa90386-1a39-4518-9286-7c5ce1eb7485)